client.go

```diff
@@ -294,22 +305,30 @@ func (c *Client) AddIdempotencyHeader(req *http.Request, idempotencyKey string)
 }
 
 func (c *Client) Do(req *http.Request, successResponse, errorResponse interface{}) (*Response, error) {
+	_, span := c.newSpan(
+		req.Context(),
+		"merchant openbank client request",
+		trace.SpanKindClient,
+	)
+	defer c.endSpan(span)
+
+	c.addSpanAttribute(span, attribute.String("http.request.path", req.URL.String()))
+	c.addSpanAttribute(span, attribute.String("http.request.protocol", req.Proto))
+	c.addSpanAttribute(span, attribute.String("http.request.method", req.Method))
+
 	if c.debug {
 		d, _ := httputil.DumpRequestOut(req, true)
 		c.log.Infof(">>> REQUEST:\n%s", string(d))
 	}
 
-	txn := newrelic.FromContext(req.Context())
-	defer newrelic.StartSegment(txn, "merchant openbank client request").End()
-
-	s := newrelic.StartExternalSegment(txn, req)
-	defer s.End()
-
 	resp, err := c.client.Do(req)
 	if err != nil {
+		c.setSpanStatus(span, codes.Error, "error executing request")
```

Comment:
Acho que seria interessante adicionar um RecordError aqui também
```suggestion
		c.setSpanStatus(span, codes.Error, "error executing request")
		c.spanRecordError(span, err)
```